a4kSubtitles/download.py

```diff
@@ -67,9 +76,8 @@ def __extract_zip(core, archivepath, filename, episodeid):
     return dest
 
 def __insert_lang_code_in_filename(core, filename, lang_code):
-    filename_chunks = core.utils.strip_non_ascii_and_unprintable(filename).split('.')
-    filename_chunks.insert(-1, lang_code)
-    return '.'.join(filename_chunks)
+    fn = core.utils.strip_non_ascii_and_unprintable(filename).rsplit(".", 1)[0]
```

Comment:
Could you add example what we accomplish with this change?
Also fn is often used as short for function, it is a bit confusing for name here.

Reply:
The old method assumes that there always "." in the filename and it's ends with file extension which sometimes it doesn't. I noticed this It's sometimes I saw Kodi doesn't shows my lang_code correctly.

Here is an example tests with old. you may notice it doesn't insert the lang_code correctly due to sometimes "dot" may not be at the end or translator website may be int or audio type

<details><summary>Old method</summary>
<p>

```
[[ Filename: Series Episode 04x02 ]] 
[[ Chunks: ['ara', 'Series Episode 04x02'] ]]
----------------
[[ Filename: Series.2009.S01E14.English.HDTV.XviD-LOL ]] 
[[ Chunks: ['Series', '2009', 'S01E14', 'English', 'HDTV', 'ara', 'XviD-LOL'] ]]
----------------
[[ Filename: Series 2009 S01E14 English HDTV XviD LOL ]] 
[[ Chunks: ['ara', 'Series 2009 S01E14 English HDTV XviD LOL'] ]]
----------------
[[ Filename: Series Now S05 EXTRAS DVDRip XviD RUNNER ]] 
[[ Chunks: ['ara', 'Series Now S05 EXTRAS DVDRip XviD RUNNER'] ]]
----------------
[[ Filename: Series.Title.S01.EXTRAS.DVDRip.XviD-RUNNER ]] 
[[ Chunks: ['Series', 'Title', 'S01', 'EXTRAS', 'DVDRip', 'ara', 'XviD-RUNNER'] ]]
----------------
[[ Filename: 2020.Series.2011.12.02.PDTV.XviD-C4TV ]] 
[[ Chunks: ['2020', 'Series', '2011', '12', '02', 'PDTV', 'ara', 'XviD-C4TV'] ]]
----------------
[[ Filename: The.Series.S03E115.DVDRip.XviD-OSiTV ]] 
[[ Chunks: ['The', 'Series', 'S03E115', 'DVDRip', 'ara', 'XviD-OSiTV'] ]]
----------------
[[ Filename: Series Title - S01E01 - Pilot [HTDV-480p] ]] 
[[ Chunks: ['ara', 'Series Title - S01E01 - Pilot [HTDV-480p]'] ]]
----------------
[[ Filename: Series Title - S01E01 - Pilot [HTDV-720p] ]] 
[[ Chunks: ['ara', 'Series Title - S01E01 - Pilot [HTDV-720p]'] ]]
----------------
[[ Filename: Series Title - S01E01 - Pilot [HTDV-1080p] ]] 
[[ Chunks: ['ara', 'Series Title - S01E01 - Pilot [HTDV-1080p]'] ]]
----------------
[[ Filename: The.Series.Title.S04E13.720p.WEB-DL.AAC2.0.H.264-Cyphanix ]] 
[[ Chunks: ['The', 'Series', 'Title', 'S04E13', '720p', 'WEB-DL', 'AAC2', '0', 'H', 'ara', '264-Cyphanix'] ]]
----------------
[[ Filename: Series.S02E01.720p.WEB-DL.DD5.1.H.264.mkv ]] 
[[ Chunks: ['Series', 'S02E01', '720p', 'WEB-DL', 'DD5', '1', 'H', '264', 'ara', 'mkv'] ]]
----------------
[[ Filename: Series Title S01E01 Episode Title ]] 
[[ Chunks: ['ara', 'Series Title S01E01 Episode Title'] ]]
----------------
[[ Filename: The Series Title - 2014-06-02 - Thomas Piketty.mkv ]] 
[[ Chunks: ['The Series Title - 2014-06-02 - Thomas Piketty', 'ara', 'mkv'] ]]
----------------
[[ Filename: The Series Title S12E17 May 23 ]] 
[[ Chunks: ['ara', 'The Series Title S12E17 May 23'] ]]
----------------
[[ Filename: Reizen Waes - S01E08 - Transistri ]] 
[[ Chunks: ['ara', 'Reizen Waes - S01E08 - Transistri'] ]]
----------------
[[ Filename: The Series Title 10x11 - Wild Devs Cant Be Broken [rl].avi ]] 
[[ Chunks: ['The Series Title 10x11 - Wild Devs Cant Be Broken [rl]', 'ara', 'avi'] ]]
----------------
[[ Filename: [ www.Torrenting.com ] - Series.S03E14.720p.HDTV.X264-DIMENSION ]] 
[[ Chunks: ['[ www', 'Torrenting', 'com ] - Series', 'S03E14', '720p', 'HDTV', 'ara', 'X264-DIMENSION'] ]]
----------------
[[ Filename: Series S02E09 HDTV x264-2HD [eztv]-[rarbg.com] ]] 
[[ Chunks: ['Series S02E09 HDTV x264-2HD [eztv]-[rarbg', 'ara', 'com]'] ]]
----------------
[[ Filename: 7s-Series-s02e01-720p.mkv ]] 
[[ Chunks: ['7s-Series-s02e01-720p', 'ara', 'mkv'] ]]
----------------
[[ Filename: The.Series.S09E13.720p.HEVC.x265-MeGusta-Pre ]] 
[[ Chunks: ['The', 'Series', 'S09E13', '720p', 'HEVC', 'ara', 'x265-MeGusta-Pre'] ]]
----------------
[[ Filename: Series Title - S01E01 - Episode Title [RlsGroup] ]] 
[[ Chunks: ['ara', 'Series Title - S01E01 - Episode Title [RlsGroup]'] ]]
----------------
[[ Filename: Red Series S01 E01-E02 1080p AMZN WEBRip DDP5.1 x264 monkee ]] 
[[ Chunks: ['Red Series S01 E01-E02 1080p AMZN WEBRip DDP5', 'ara', '1 x264 monkee'] ]]
----------------
[[ Filename: Series.Title.S01E05.The-Aniversary.WEBDL-1080p.mkv ]] 
[[ Chunks: ['Series', 'Title', 'S01E05', 'The-Aniversary', 'WEBDL-1080p', 'ara', 'mkv'] ]]
----------------
[[ Filename: Series.Title.S01E05.The-Aniversary.HDTV-1080p.mkv ]] 
[[ Chunks: ['Series', 'Title', 'S01E05', 'The-Aniversary', 'HDTV-1080p', 'ara', 'mkv'] ]]
----------------
[[ Filename: Series US (2010) S04 (1080p BDRip x265 10bit DTS-HD MA 5 1 - WEM)[TAoE] ]] 
[[ Chunks: ['ara', 'Series US (2010) S04 (1080p BDRip x265 10bit DTS-HD MA 5 1 - WEM)[TAoE]'] ]]
----------------
[[ Filename: The.Series.S03E04.2160p.Amazon.WEBRip.DTS-HD.MA.5.1.x264 ]] 
[[ Chunks: ['The', 'Series', 'S03E04', '2160p', 'Amazon', 'WEBRip', 'DTS-HD', 'MA', '5', '1', 'ara', 'x264'] ]]
----------------
[[ Filename: SomeShow.S20E13.1080p.BluRay.DTS-X.MA.5.1.x264 ]] 
[[ Chunks: ['SomeShow', 'S20E13', '1080p', 'BluRay', 'DTS-X', 'MA', '5', '1', 'ara', 'x264'] ]]
----------------
[[ Filename: SomeShow.S20E13.1080p.BluRay.DTS-MA.5.1.x264 ]] 
[[ Chunks: ['SomeShow', 'S20E13', '1080p', 'BluRay', 'DTS-MA', '5', '1', 'ara', 'x264'] ]]
----------------
[[ Filename: SomeShow.S20E13.1080p.Blu-Ray.DTS-ES.5.1.x264 ]] 
[[ Chunks: ['SomeShow', 'S20E13', '1080p', 'Blu-Ray', 'DTS-ES', '5', '1', 'ara', 'x264'] ]]
----------------
[[ Filename: SomeShow.S20E13.1080p.Blu-Ray.DTS-ES.5.1.x264-ROUGH [PublicHD] ]] 
[[ Chunks: ['SomeShow', 'S20E13', '1080p', 'Blu-Ray', 'DTS-ES', '5', '1', 'ara', 'x264-ROUGH [PublicHD]'] ]]
----------------
[[ Filename: SomeShow S01E168 1080p WEB-DL AAC 2.0 x264-Erai-raws ]] 
[[ Chunks: ['SomeShow S01E168 1080p WEB-DL AAC 2', 'ara', '0 x264-Erai-raws'] ]]
----------------
[[ Filename: The.Good.Series.S05E03.Series.of.Intelligence.1080p.10bit.AMZN.WEB-DL.DDP5.1.HEVC-Vyndros ]] 
[[ Chunks: ['The', 'Good', 'Series', 'S05E03', 'Series', 'of', 'Intelligence', '1080p', '10bit', 'AMZN', 'WEB-DL', 'DDP5', '1', 'ara', 'HEVC-Vyndros'] ]]
----------------
[[ Filename: [Tenrai-Sensei] Series [BD][1080p][HEVC 10bit x265 ][AAC 2.0][Multi-Subs] ]] 
[[ Chunks: ['[Tenrai-Sensei] Series [BD][1080p][HEVC 10bit x265 ][AAC 2', 'ara', '0][Multi-Subs]'] ]]
----------------
[[ Filename: [Tenrai-Sensei] Series [BD][1080p][HEVC 10bit x265][AAC 2.0][Multi-Subs][Tenrai-Sensei] ]] 
[[ Chunks: ['[Tenrai-Sensei] Series [BD][1080p][HEVC 10bit x265][AAC 2', 'ara', '0][Multi-Subs][Tenrai-Sensei]'] ]]
----------------
[[ Filename: [Tenrai-Sensei] Series [BD][1080p][HEVC 10bit x265][AAC 2.0][Multi-Subs][Tenrai-Sensei][Tenrai-Sensei] ]] 
[[ Chunks: ['[Tenrai-Sensei] Series [BD][1080p][HEVC 10bit x265][AAC 2', 'ara', '0][Multi-Subs][Tenrai-Sensei][Tenrai-Sensei]'] ]]
----------------
[[ Filename: One.Piece.Episodes.0001-1000 ]] 
[[ Chunks: ['One', 'Piece', 'Episodes', 'ara', '0001-1000'] ]]
----------------
[[ Filename: mashle - 24 [s2 - 12 end][crunchyroll] ]] 
[[ Chunks: ['ara', 'mashle - 24 [s2 - 12 end][crunchyroll]'] ]]

```

</p>
</details> 


At start I thought we should remove the whole insert function and just go with `filename + ".{0}.format(lang_code)`
But again I thought there were a purpose of adding it but is there one?

Reply:
KODI's expected lang code format is: `filename.$lang_code.ext`
The code tried to insert it before the ext before. I guess the fix should be if ext is missing add it at the end.
The proposed solution strips the ext:
<img width="473" alt="image" src="https://github.com/user-attachments/assets/770074aa-550f-43f5-98b2-83574450c163">

Reply:
But mostly at the point of calling lang code insert we still don't know what is the subtitle extension yet filename refers to the subtitle release name / archive name at the end we may end with `filename.$lang_code.$media_ext.$subtitle_ext` Before Kodi were copying the subtitles with the media extension to me if archive.

Reply:
Should we go with this solution for now?
```py
def __insert_lang_code_in_filename(core, filename, lang_code):
    name = core.utils.strip_non_ascii_and_unprintable(filename)
    nameparts = name.rsplit(".", 1)

    # Because this can be called via "raw" subtitles where sub ext exists we will ensure it ends with the subtitle ext.
    # Otherwise we will use "filename.lang_code" later the ext will be added on unzip process.
    if len(nameparts) > 1 and ("." + nameparts[1] in subtitles_exts + subtitles_exts_secondary):
        return ".".join([nameparts[0], lang_code, nameparts[1]])

    return "{0}.{1}".format(name, lang_code)
```